modules/dynamic-plug-in-api.adoc

```diff
@@ -0,0 +1,1467 @@
+
+[source,tsx]
+----
+   const logNamespaceChange = (namespace) => console.log(`New namespace: ${namespace}`);
+
+   ...
+
+   <NamespaceBar onNamespaceChange={logNamespaceChange}>
+     <NamespaceBarApplicationSelector />
+   </NamespaceBar>
+   <Page>
+
+     ...
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`onNamespaceChange` |(optional) A function that is executed when a
+namespace option is selected. It accepts the new namespace in the form
+of a string as its only argument. The active namespace is updated
+automatically when an option is selected, but additional logic can be
+applied via this function. When the namespace is changed, the namespace
+parameter in the URL will be changed from the previous namespace to the
+newly selected namespace.
+
+|`isDisabled` |(optional) A boolean flag that disables the namespace
+dropdown if set to true. This option only applies to the namespace
+dropdown and has no effect on child components.
+
+|`children` |(optional) Additional elements to be rendered inside the
+toolbar to the right of the namespace dropdown.
+|===
+
+[discrete]
+== `ErrorBoundaryFallbackPage`
+
+Creates full page ErrorBoundaryFallbackPage component to display the "Oh no! Something went wrong." message along with the stack trace and other helpful debugging information. This is to be used inconjunction with an component.
+
+[source,tsx]
+----
+//in ErrorBoundary component
+ return (
+   if (this.state.hasError) {
+     return <ErrorBoundaryFallbackPage errorMessage={errorString} componentStack={componentStackString}
+      stack={stackTraceString} title={errorString}/>;
+   }
+
+   return this.props.children;
+ }
+)
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`errorMessage` |text description of the error message
+|`componentStack` |component trace of the exception
+|`stack` |stack trace of the exception
+|`title` |title to render as the header of the error boundary page
+|===
+
+[discrete]
+== `PerspectiveContext`
+
+@deprecated - use the provided `usePerspectiveContext` insteadCreates the perspective context.
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`PerspectiveContextType` |object with active perspective and setter
+|===
+
+[discrete]
+== `useAccessReviewAllowed`
+
+@deprecated - Use useAccessReview from @console/dynamic-plugin-sdk instead.Hook that provides allowed status about user access to a given
+resource. It returns the `isAllowed` boolean value.
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`resourceAttributes` |resource attributes for access review
+|`impersonate` |impersonation details
+|===
+
+[discrete]
+== `useSafetyFirst`
+
+@deprecated - This hook is not related to console functionality.Hook that ensures a safe asynchronnous setting of React state in case a given
```

Comment:
S/functionality.Hook/functionality. Hook 
S/React state/the React state ??
~~(https://github.com/facebook/react/issues/14113)~~